### DISCUSSION TITLE
Staying up to date with Nightly builds?

### DISCUSSION THREAD

**Discussion by CodyPrince** [OrcaSlicer/OrcaSlicer#1575] [Dev]
I love the development Ocra has, so I'm constantly checking the "Actions" page for updated "Nightlys". On the actions page, left side, I select "Build Win64". Then I select the update, or I think better called "workflow", that I want to test. If it has a "OrcaSlicer_Win64_nightly" in the bottom under artifacts, Then I download it and add to the name of the file the name of the update. For example I have one folder called OrcaSlicer_Win64_nightly_Ellis, one folder called OrcaSlicer_Win64_nightly_Overhangs, and one folder called OrcaSlicer_Win64_nightly_Lift Z. I then use each of the different versions depending on what I want to do. 

Question, Is there a way to run all these great new features at one time in one version of orca? I don't know if its "building" and "compiling", but I want a beta version with all the latest updates. 

I hope my question makes sense, Cody

**Comment by SoftFever** [comment]
The best way is to download the main branch’s nightly build as all PR will be merged into main branch when it’s ready.
We can probably post the download link of latest nightly build somewhere.
Let me think about it. If you have any suggestions, please let me know too.

**Reply by CodyPrince** [reply]
I can not find where to download the "Main Branch Nightly Build" anywhere. Posting a link would be great, but not necessary if someone could just tell me where to download. 

I think a good place if you were to post a link would be on the main page where you talk about the Prebuilt binaries are available through the github releases page.

**Reply by SoftFever** [reply]
I see. Hope this will help you
![nightly](https://github.com/SoftFever/OrcaSlicer/assets/103989404/5cbde38b-f78b-451a-ac14-b85b6f86d500)

**Reply by CodyPrince** [reply]
That worked. I've got all the new features I was wanting except "make overhangs printable". I guess because it doesn't have the "main" at the right of workflow. 

I'm checking daily, so I will just watch for it. Looking forward to single wall on top surface fix while using classic walls as well as extrusion width as percent. Looking forward to v1.6.4.

**Comment by Waffull** [comment]
Wow, I didn't realize nightly builds were available.  I've been compiling them myself when I wanted to test something out.  This is sooooo much easier :)

Thank you for exposing this option in github :)

**Reply by RaySajuuk** [reply]
I prefer build it myself, also help to create universal binary and notarize it. So it can be opened on any modern macOS without all the workarounds.

**Comment by dlarue** [comment]
When I select "Actions" in github I don't see those OS options in OrcaSlicer so I can't find the latest beta to try OrcaSlicer on my Linux machines. Any other way to find the links to the daily builds/betas?

**Reply by CodyPrince** [reply]
They changed thing around a few weeks ago. I was going to post the same question this weekend. I'm unsure how to do get them now as well.

**Reply by dlarue** [reply]
Thanks @CodyPrince the post was so explicit by showing the process of getting to the builds I though my permissions on github were preventing me from seeing what I was supposed to see.  Maybe @SoftFever can update this and if not the graphics, just a few lines of steps as to where to find it.  I'm donating a Prusa MK3.5 to a local school who has 2 MakerBots Rep+ units and while I can setup Prusa Slicer for them I wanted to see if OrcaSlicer is all people are saying it is.  Robotics.... so Linux.

**Reply by darco415** [reply]
The builds are still there. Action/ click on link next to green check mark...build 1.7 dev currently showing

**Reply by dlarue** [reply]
I guess I'm just dense.... Clicked "Action" options and there are a bunch of links in the right list window area with green check marks but none have any label close to "build 1.7 dev".  Nothing even close in the left l list area.
![action-screen](https://github.com/SoftFever/OrcaSlicer/assets/3150680/796ce6c2-91c9-4b1f-9080-ced09998686c)

**Reply by darco415** [reply]
just click on the link "initial port...." and it will take you to the builds

**Reply by CodyPrince** [reply]
The "initial Port....", which is currently the latest, is not part of the main branch. I don't want to lose other build versions just to test one version. Do I still need to download only the "Main" version like softfever mentioned earlier in this post?

Thanks, Cody

**Reply by darco415** [reply]
that link was just an example just to show the builds are still out there. I would assume the Main branch are the builds you want.

**Comment by dlarue** [comment]
> just click on the link "initial port...." and it will take you to the builds

Thanks, it wasn't the top/latest "initial port...." as that only had the build output with no artifacts listed. I finally tried yesterdays 'initial port..." and there is an artifacts section with a Ubuntu zip file containing and AppImage file. Trying it now.

**Comment by EBADBEEF** [comment]
Every time a commit is made to the main branch, a build under the "Build all" action is done. That becomes the most up-to-date build.
Here is a link to all the builds that successfully completed on the main branch: https://github.com/SoftFever/OrcaSlicer/actions/workflows/build_all.yml?query=branch%3Amain+is%3Asuccess

Note that the branch search is not exact, so you should probably pick one that says just "main".

**Comment by akrtek** [comment]
Sorry this is a learning question. I have previously used the github desktop to clone repos on projects which I have done here, but I am not sure how to locate the files found under the actions tab on my cloned repo. thought it would be easier to update nightlys this way.
